lineman/app/services/thread_query_service.coffee

```diff
@@ -1,49 +1,40 @@
 angular.module('loomioApp').factory 'ThreadQueryService', (Records) ->
   new class ThreadQueryService
 
-    filterQuery: (filter) ->
-      threadQueryFor createBaseView(), filter
-
-    groupQuery: (group = {}, options = {}) ->
-      threadQueryFor createGroupView(group), 
-                     options['filter'] or 'show_unread',
-                     options['limit'] or 5
+    filterQuery: (filter, options = {}) ->
+      threadQueryFor createBaseView(filter, options['queryType'] or 'all')
 
     timeframeQuery: (options = {}) ->
-      threadQueryFor createTimeframeView(options['name'], options['timeframe']),
-                     options['filter'] or 'show_all',
-                     options['limit'] or 100
+      threadQueryFor createTimeframeView(options['name'], options['filter'] or 'show_all', 'timeframe', options['timeframe']['from'], options['timeframe']['to'])
```

Comment:
I've simplified this class a lot. Now it is much easier to understand.

Now we're finally applying conditions to a loki view directly based on the query type and filters you pass in.